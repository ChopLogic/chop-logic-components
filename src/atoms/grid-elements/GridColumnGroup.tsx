import React from 'react';
import { ChopLogicGridColumn } from '@models';
import styles from '@organisms/grid/Grid.module.scss';
import { getClassName } from '@utils';

export const GridColumnGroup: React.FC<{ columns: ChopLogicGridColumn[]; selectable: boolean }> = ({ columns, selectable }) => {
  return (
    <colgroup>
      {selectable && <col className={getClassName([styles.grid_column, styles.grid_column__first])} />}
      {columns.map((column, index) => (
        <col
          key={column?.title}
          className={getClassName([
            styles.grid_column,
            column?.className,
            {
              [styles.grid_column__first]: index === 0 && !selectable,
              [styles.grid_column__highlighted]: !!column?.highlighted,
            },
          ])}
        />
      ))}
    </colgroup>
  );
};
