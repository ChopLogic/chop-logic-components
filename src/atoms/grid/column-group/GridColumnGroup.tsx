import { ChopLogicGridColumn } from '@models';
import { getClassName } from '@utils';
import React from 'react';

import styles from './GridColumnGroup.module.scss';

export const GridColumnGroup: React.FC<{ columns: ChopLogicGridColumn[]; selectable: boolean }> = ({ columns, selectable }) => {
  return (
    <colgroup>
      {selectable && <col className={getClassName([styles.column, styles.column__first])} />}
      {columns.map((column, index) => (
        <col
          key={column?.title}
          className={getClassName([
            styles.column,
            column?.className,
            {
              [styles.column__first]: index === 0 && !selectable,
              [styles.column__highlighted]: !!column?.highlighted,
            },
          ])}
        />
      ))}
    </colgroup>
  );
};
