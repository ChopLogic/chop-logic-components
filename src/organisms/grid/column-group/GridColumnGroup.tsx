import { GridColumn } from '@models';
import { getClassName } from '@utils';
import { FC } from 'react';

import styles from './GridColumnGroup.module.scss';

export const GridColumnGroup: FC<{ columns: GridColumn[]; selectable: boolean }> = ({ columns, selectable }) => {
  return (
    <colgroup>
      {selectable && <col className={getClassName([styles.column, styles.column__first])} />}
      {columns.map((column, index) => (
        <col
          key={column?.title}
          className={getClassName([
            styles.column,
            column?.className,
            {
              [styles.column__first]: index === 0 && !selectable,
              [styles.column__highlighted]: !!column?.highlighted,
            },
          ])}
        />
      ))}
    </colgroup>
  );
};
