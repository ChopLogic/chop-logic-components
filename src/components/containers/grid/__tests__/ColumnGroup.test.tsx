import { ChopLogicGridColumn } from '@models';
import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';

import GridColumnGroup from '../elements/ColumnGroup';

import { LIGHT_THEME } from '@css';

describe('ChopLogicGridBody', () => {
  const columns: ChopLogicGridColumn[] = [
    { title: 'Col 1', field: 'field1', highlighted: true },
    { title: 'Col 2', field: 'field2', className: 'column-class' },
    { title: 'Col 3', field: 'field3' },
  ];

  it('should match the snapshot', () => {
    const { asFragment } = render(
      <table>
        <GridColumnGroup columns={columns} selectable theme={LIGHT_THEME} />
      </table>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
