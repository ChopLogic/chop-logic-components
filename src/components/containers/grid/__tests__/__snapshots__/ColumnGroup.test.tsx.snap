// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ChopLogicGridBody > should match the snapshot 1`] = `
<DocumentFragment>
  <table>
    <colgroup>
      <col
        class="sc-hRDKVd gVggjg"
      />
      <col
        class="sc-hRDKVd gwPyPd"
      />
      <col
        class="sc-hRDKVd dIOaAQ column-class"
      />
      <col
        class="sc-hRDKVd dIOaAQ"
      />
    </colgroup>
  </table>
</DocumentFragment>
`;
